index.js

```diff
@@ -243,6 +246,79 @@ module.exports = function(version, _options) {
             }
 
             return output;
+        },
+        getBestMatchingLanguage: function(language) {
+            if (languages.instructions[language]) return language;
+            var that = this;
+
+            var codes = that.divideLanguageCodes(language);
+            var languageCode = codes.languageCode;
+            var scriptCode = codes.scriptCode;
+            var countryCode = codes.countryCode;
+
+            var supportedLanguageCodes = languages.supportedCodes.map(function(language) {
+                return language.toLowerCase().split('-')[0];
+            });
+
+            var availableLanguages = languages.supportedCodes.map(function(language) {
+                return that.divideLanguageCodes(language);
+            });
+
+            // Same language code and script code (lng-Scpt)
+            if (languages.instructions[languageCode + '-' + scriptCode]) {
+                return languageCode + '-' + scriptCode;
+            // Same language code and country code (lng-CC)
+            } else if (languages.instructions[languageCode + '-' + countryCode]) {
+                return languageCode + '-' + countryCode;
+            // Same language code (lng)
+            } else if (languages.instructions[languageCode]) {
+                return languageCode;
+            // Same language code and any script code (lng-Scpx) and the found language contains a script
+            } else if (availableLanguages.find(function (language) {
+                return language.languageCode === languageCode && language.scriptCode;
+            })) {
+                return languages.supportedCodes[supportedLanguageCodes.indexOf(languageCode)];
+            // Same language code and any country code (lng-CX)
+            } else if (supportedLanguageCodes.indexOf(languageCode) > -1 && countryCode) {
+                return languages.supportedCodes[supportedLanguageCodes.indexOf(languageCode)];
+            } else {
+                return 'en';
+            }
+        },
+        divideLanguageCodes: function(language) {
+            // If the language is not found, try a little harder
+            var splitLocale = language.toLowerCase().split('-');
```

Comment:
To me, it’d be cleaner to use a regular expression for this job:

```js
// Two-letter language code, then optional four-letter script code, then optional two-letter region code
var match = language.match(/(\w\w)(?:-(\w\w\w\w))?(?:-(\w\w))?/i);
return {
  locale: match[0],
  language: match[1],
  script: match[2],
  region: match[3]
};
```

Reply:
This is failing for me. I think both styles of parsing out the codes is fine.

Reply:
If it’s failing it’s probably because the rest of the code hasn’t been updated to account for the normalized case being used by ☝️.